dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/tests/BaggageTest.java

```diff
@@ -0,0 +1,114 @@
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.impl.LibertyServerFactory;
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+
+@RunWith(FATRunner.class)
+@Mode(TestMode.LITE)
+public class BaggageTest extends MicroProfileTelemetryTestBase {
+
+    private static Class<?> c = AutoInstrumentationTest.class;
+    private static LibertyServer server = LibertyServerFactory.getLibertyServer("testServer1");
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+
+        String os = System.getProperty("os.name").toLowerCase();
+        Log.info(c, "setUp", "os.name = " + os);
+
+        clearContainerOutput();
+        String jaegerHost = jaegerContainer.getHost();
+        String jaegerGrpcPort = String.valueOf(jaegerContainer.getMappedPort(JAEGER_GRPC_PORT));
+        Log.info(c, "setUp", "Jaeger container: host=" + jaegerHost + "  port=" + jaegerGrpcPort);
+        server.addEnvVar(ENV_OTEL_SERVICE_NAME, OTEL_SERVICE_NAME_SYSTEM);
+        server.addEnvVar(ENV_OTEL_TRACES_EXPORTER, OTEL_TRACES_EXPORTER_JAEGER);
+        server.addEnvVar(ENV_OTEL_EXPORTER_JAEGER_ENDPOINT, "http://" + jaegerHost + ":" + jaegerGrpcPort);
+        server.addEnvVar(ENV_OTEL_SDK_ENABLED, OTEL_SDK_ENABLED); // tracing enabled
+        // Construct the test application
+        WebArchive system = ShrinkWrap.create(WebArchive.class, "system.war");
+        system.addPackages(true, "io.openliberty.guides.system");
+        ShrinkHelper.exportAppToServer(server, system);
+        serverStart();
+    }
+
+    @Before
+    public void setUpTest() throws Exception {
+
+        if (!server.isStarted()) {
+            serverStart();
+        }
+        server.addInstalledAppForValidation("system");
+    }
+
+    @After
+    public void tearDown() {
+    }
+
+    @AfterClass
+    public static void completeTest() throws Exception {
+        try {
+            if (server.isStarted()) {
+                Log.info(c, "competeTest", "---> Stopping server..");
+                server.stopServer("TRAS4301W");
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void serverStart() throws Exception {
+        Log.info(c, "serverStart", "--->  Starting Server.. ");
+        server.startServer();
+    }
+
+    @Override
+    protected LibertyServer getServer() {
+        return server;
+    }
+
+    @Test
+    void current_empty() {
+        try (Scope scope = Context.root().makeCurrent()) {
+            assertEquals(Baggage.current(), Baggage.empty());
+        }
+    }
+
+    @Test
+    void current() {
+        try (Scope scope = Context.root().with(Baggage.builder().put("foo", "bar").build()).makeCurrent()) {
+            Baggage result = Baggage.current();
+            assertEquals(result.getEntryValue("foo"), "bar");
+        }
+    }
```

Comment:
I think these tests should run on the server?

Putting them here they'll run outside the server. We need to create a test servlet and move the tests there.